pex/pex_info.py

```diff
@@ -60,24 +60,9 @@ class PexInfo(object):
     INSTALL_CACHE = "installed_wheels"
 
     @classmethod
-    def make_build_properties(cls, interpreter=None):
-        # This lazy import is currently needed for performance reasons. At PEX runtime PexInfo is
-        # read in the bootstrap to see if the PEX should run in `--unzip` mode. If so, it must
-        # re-exec itself to run against its unzipped contents. Since `make_build_properties` is only
-        # used at PEX buildtime and the transitive imports of PythonInterpreter are large and slow,
-        # we avoid this import cost for runtime-only use.
-        #
-        # See: https://github.com/pantsbuild/pex/issues/1054
-        from pex.interpreter import PythonInterpreter
-
-        pi = interpreter or PythonInterpreter.get()
-        plat = pi.platform
-        platform_name = plat.platform
+    def make_build_properties(cls):
         return {
             "pex_version": pex_version,
-            "class": pi.identity.interpreter,
-            "version": pi.identity.version,
-            "platform": platform_name,
```

Comment:
Pex still has no API except the CLI.

Reply:
Agreed there's no deprecation policy and this API is seen as unstable. Although, imo this isn't binary either: you can still take steps to make changes more transparent. Pants's plugin API is not stable either, but we still try to make plugin API changes into dedicated PRs so that they show up in the changelog. 

Here, it seems valuable to have a changelog entry: "Remove class, version, and platform from PEX-INFO".

Reply:
That's only valuable if you want to tacitly assume / be an enabler for illicit API usage though! I definitely don't want to do that. If you use the code APIs I want you to learn fast not to do that!

Reply:
The rules API is a poor comparison since you _do_ want folks to use that.

Reply:
This helped expose a Pants bug as it turns out!